### PR TITLE
CI Workflow isn't testing sub-crates

### LINKED ISSUE #16: CI Workflow isn't testing sub-crates
The crate `vmm` in `src/vmm` isn't tested when we run our CI Workflow, it's a problem as there are tests in this crate. Update the CI so the `vmm` crate is tested as well.

### PR DESCRIPTION
**Pull request by NelopsisCode**
close #16

### CODE REVIEW

File: .github/workflows/lumper.yml [hunk 1]
```diff
@@ -28,9 +28,7 @@ jobs:
           components: rustfmt, clippy
 
       - name: Run cargo test
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
+        run: cargo test && cargo test -p vmm
```
Comment (sameo) [inline .github/workflows/lumper.yml:31]: Can you try a `cargo test --all`?

Comment (NelopsisCode) [reply .github/workflows/lumper.yml:31]: I tried, it not run test for crate vmm. `cargo test --workspace` either...

Comment (AlexandreBrg) [reply .github/workflows/lumper.yml:31]: As `vmm` is a crate inside another one, it's not a workspace, so we can't add `--workspace`. We might need to move `vmm` at the workspace level

Comment (sameo) [reply .github/workflows/lumper.yml:31]: You're both right, thanks.

We should remove the kernel stuff under a `data` folder and and the rootfs build script under a `hacks`one. Then `mv src/vmm vmm`. I'll take care of that once all pending PRs are merged, and I'm merging that one now.

**Review by sameo — APPROVED** [review]

**Pull request merged**